h2/src/main/org/h2/command/Parser.java

```diff
@@ -8626,7 +8629,68 @@ private Table readTableOrView(String tableName) {
         if (isDualTable(tableName)) {
             return new DualTable(database);
         }
-        throw DbException.get(ErrorCode.TABLE_OR_VIEW_NOT_FOUND_1, tableName);
+
+        throw getTableOrViewNotFoundDbException(tableName);
+    }
+
+    private DbException getTableOrViewNotFoundDbException(final String tableName) {
+        if (schemaName != null) {
+            return getTableOrViewNotFoundDbException(schemaName, tableName);
+        }
+
+        final String currentSchemaName = session.getCurrentSchemaName();
+        final String[] schemaSearchPath = session.getSchemaSearchPath();
+        if (schemaSearchPath == null) {
+            return getTableOrViewNotFoundDbException(Collections.singleton(currentSchemaName), tableName);
+        }
+
+        final LinkedHashSet<String> schemaNames = new LinkedHashSet<>();
+        schemaNames.add(currentSchemaName);
+        schemaNames.addAll(Arrays.asList(schemaSearchPath));
+        return getTableOrViewNotFoundDbException(schemaNames, tableName);
+    }
+
+    private DbException getTableOrViewNotFoundDbException(final String schemaName, final String tableName) {
+        return getTableOrViewNotFoundDbException(Collections.singleton(schemaName), tableName);
+    }
+
+    private DbException getTableOrViewNotFoundDbException(
+            final java.util.Set<String> schemaNames, final String tableName) {
+        if (database == null || database.getFirstUserTable() == null) {
+            return DbException.get(ErrorCode.TABLE_OR_VIEW_NOT_FOUND_DATABASE_EMPTY_1, tableName);
+        }
+
+        if (database.getSettings().caseInsensitiveIdentifiers) {
+            return DbException.get(ErrorCode.TABLE_OR_VIEW_NOT_FOUND_1, tableName);
+        }
+
+        final java.util.Set<String> candidates = new TreeSet<>();
+        for (final String schemaName : schemaNames) {
+            candidates.addAll(findTableNameCandidates(schemaName, tableName));
+        }
+
```

Comment:
Pass `candidates` to `findTableNameCandidates()` instead of allocation of own set it in.